pkg/asset/installconfig/kubevirt/validation.go

```diff
@@ -0,0 +1,120 @@
+
+	client, resultErrs := validateInfraClusterReachable(kubevirtPlatform.Namespace, fldPath, clientBuilder)
+	allErrs = append(allErrs, resultErrs...)
+	if client != nil {
+		allErrs = append(allErrs, validateStorageClassExistsInInfraCluster(kubevirtPlatform.StorageClass, client, fldPath)...)
+		allErrs = append(allErrs, validateNetworkAttachmentDefinitionExistsInNamespace(kubevirtPlatform.NetworkName, kubevirtPlatform.Namespace, client, fldPath)...)
+	}
+	allErrs = append(allErrs, validateIPsInMachineNetworkEntryList(machineNetworkEntryList, kubevirtPlatform.APIVIP, kubevirtPlatform.IngressVIP, fldPath)...)
+
+	return allErrs
+}
+
+// validateInfraClusterReachable validates the following:
+// 1. Client can be created -  login to cluster is done or KUBECONFIG environment variable was exported
+// 2. Cluster is reachable - valid user
+// 3. Kubevirt installed
+// 4. User can list VMs in namespace
+func validateInfraClusterReachable(namespace string, fieldPath *field.Path, clientBuilder ClientBuilderFuncType) (Client, field.ErrorList) {
+	allErrs := field.ErrorList{}
+	client, err := clientBuilder()
+	var detailedErr error
+	if err != nil {
+		detailedErr = fmt.Errorf("failed to create InfraCluster client with error: %v", err)
+	} else if _, err := client.ListVirtualMachine(namespace, metav1.ListOptions{}); err != nil {
+		detailedErr = fmt.Errorf("failed to list VMs with error: %v", err)
+	}
+
+	if detailedErr != nil {
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("InfraClusterReachable"),
+			"InfraCluster",
+			detailedErr.Error(),
+		))
+		return nil, allErrs
+	}
+
+	return client, allErrs
+}
+
+func validateStorageClassExistsInInfraCluster(name string, client Client, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if _, err := client.GetStorageClass(name); err != nil {
+		detailedErr := fmt.Errorf("failed to get StorageClass %s from InfraCluster, with error: %v", name, err)
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("StorageClassExistsInInfraCluster"), name, detailedErr.Error()))
+	}
+
+	return allErrs
+}
+
+// validateNetworkAttachmentDefinitionExistsInNamespace validate the following:
+// 1. The namespace exists
+// 2. The network-attachment-definition exists
+func validateNetworkAttachmentDefinitionExistsInNamespace(name string, namespace string, client Client, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if _, err := client.GetNetworkAttachmentDefinition(name, namespace); err != nil {
+		detailedErr := fmt.Errorf("failed to get network-attachment-definition %s from InfraCluster, with error: %v", name, err)
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("NetworkAttachmentDefinitionExistsInInfraCluster"), name, detailedErr.Error()))
+	}
+
+	return allErrs
+}
+
+func validateIPsInMachineNetworkEntryList(machineNetworkEntryList []types.MachineNetworkEntry, apiVIP string, ingressVIP string, fieldPath *field.Path) field.ErrorList {
```

Comment:
This validation should be in `pkg/types/kubevirt/validation/platform.go` as it is not dependent getting information from the server.

Reply:
Fixed